### DISCUSSION TITLE
C# - Delta is now a double instead of a float

### DISCUSSION THREAD

**Discussion by nonunknown** [godotengine/godot-proposals#5395] [Scripting]
Was actually a good Idea to convert delta from float to double? Because every function and arithmetic the user does on his code accepts only float, for example lerp function, so every time I need to convert delta to float!

this would make sense if all godot functions which takes float now takes double instead (but isnt the case)

also as far as I know double is larger than float, so why that?

**Comment by Calinou** [comment]
This was likely done to avoid downcasting issues when using `delta` in calculations, which can cause precision issues for long-running servers (> 10 days).
